it/controllers/employment/EmployerLeaveDateControllerISpec.scala

```diff
@@ -0,0 +1,927 @@
+            textOnPageCheck(expectedCaption(taxYearEOY), captionSelector)
+            textOnPageCheck(forExample, forExampleSelector)
+            inputFieldValueCheck("05", Selectors.daySelector)
+            inputFieldValueCheck("04", Selectors.monthSelector)
+            inputFieldValueCheck((taxYearEOY-1).toString, Selectors.yearSelector)
+            buttonCheck(expectedButtonText, continueButtonSelector)
+            formPostLinkCheck(continueLink, continueButtonFormSelector)
+            welshToggleCheck(user.isWelsh)
+
+            errorSummaryCheck(user.specificExpectedResults.get.tooLongAgoDateError, Selectors.daySelector)
+            errorAboveElementCheck(user.specificExpectedResults.get.tooLongAgoDateError, Some("amount"))
+          }
+
+          "the date is not in the past" which {
+            val nowDatePlusOne = LocalDate.now().plusDays(1)
+            lazy val form: Map[String, String] = Map(
+              EmploymentDateForm.year -> nowDatePlusOne.getYear.toString,
+              EmploymentDateForm.month -> nowDatePlusOne.getMonthValue.toString,
+              EmploymentDateForm.day -> nowDatePlusOne.getDayOfMonth.toString)
+
+            lazy val result: WSResponse = {
+              dropEmploymentDB()
+              insertCyaData(employmentUserData(isPrior = true, cyaModel(employerName, hmrc = true)), userRequest)
+              authoriseAgentOrIndividual(user.isAgent)
+              urlPost(employerEndDatePageUrl(taxYearEOY), body = form, follow = false, welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+            }
+
+            "has the correct status" in {
+              result.status shouldBe BAD_REQUEST
+            }
+
+            implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+            import Selectors._
+            import user.commonExpectedResults._
+
+            titleCheck(user.specificExpectedResults.get.expectedErrorTitle)
+            h1Check(user.specificExpectedResults.get.expectedH1)
+            textOnPageCheck(expectedCaption(taxYearEOY), captionSelector)
+            textOnPageCheck(forExample, forExampleSelector)
+            inputFieldValueCheck(nowDatePlusOne.getDayOfMonth.toString, Selectors.daySelector)
+            inputFieldValueCheck(nowDatePlusOne.getMonthValue.toString, Selectors.monthSelector)
+            inputFieldValueCheck(nowDatePlusOne.getYear.toString, Selectors.yearSelector)
+            buttonCheck(expectedButtonText, continueButtonSelector)
+            formPostLinkCheck(continueLink, continueButtonFormSelector)
+            welshToggleCheck(user.isWelsh)
+
+            errorSummaryCheck(user.specificExpectedResults.get.futureDateError, Selectors.daySelector)
+            errorAboveElementCheck(user.specificExpectedResults.get.futureDateError, Some("amount"))
+          }
+          "the date is before the start date" which {
+            val nowDatePlusOne = LocalDate.now().plusDays(1)
```

Comment:
nowDatePlusOne isn't used - can be removed

Reply:
removed